causing/model.py

```diff
@@ -13,22 +13,25 @@ class Model:
     xvars: List[sympy.Symbol]
     yvars: List[sympy.Symbol]
     ymvars: List[sympy.Symbol]
-    equations: List[sympy.Expr]
+    equations: Tuple[sympy.Expr, ...]
     final_var: sympy.Symbol
 
     ndim: int = field(init=False)
     mdim: int = field(init=False)
 
     def __post_init__(self):
-        self.ndim = len(self.yvars)
         self.mdim = len(self.xvars)
+        self.ndim = len(self.yvars)
+
+        self.biases = [sympy.Symbol(f"bias{i+1}") for i in range(len(self.yvars))]
+        eqs_with_bias = [eq + bias for eq, bias in zip(self.equations, self.biases)]
 
-        substituted_eqs = list(self.equations)
+        substituted_eqs = list(eqs_with_bias)
         for i, yvar in enumerate(self.yvars):
             for j in range(i + 1, len(self.yvars)):
                 substituted_eqs[j] = substituted_eqs[j].subs(yvar, substituted_eqs[i])
         self.model_lam = sympy.lambdify(
-            self.xvars, substituted_eqs, modules=["sympy", "numpy"]
+            self.xvars + self.biases, substituted_eqs, modules=["sympy", "numpy"]
```

Comment:
Great how you added the biases and integrated them into the substitution. Right now I don't see how this will be called later on.